pandas/core/dtypes/astype.py

```diff
@@ -118,6 +118,8 @@ def astype_nansafe(
 
         # allow frequency conversions
         if dtype.kind == "M":
+            if dtype == "<M8[D]":
+                return arr
```

Comment:
this is going to silently return the wrong result (just a different wrong) instead of raising?

Reply:
ok. I try to use different approach.